tests/ui/generic-associated-types/issue-112895.rs

```diff
@@ -0,0 +1,29 @@
+// This test checks that we correctly reject the following unsound code.
```

Comment:
Please give this a better test name, like "gat-bounds-not-checked-with-right-substitutions.rs`